drivers/metal/rendering_device_driver_metal.mm

```diff
@@ -1536,15 +1539,19 @@ void deserialize(BufReader &p_reader) {
 };
 
 struct API_AVAILABLE(macos(11.0), ios(14.0)) ShaderBinaryData {
+	enum Flags : uint32_t {
+		NONE = 0,
+		NEEDS_VIEW_MASK_BUFFER = 1 << 0,
+		USES_ARGUMENT_BUFFERS = 1 << 1,
+	};
 	CharString shader_name;
 	// The Metal language version specified when compiling SPIR-V to MSL.
 	// Format is major * 10000 + minor * 100 + patch.
 	uint32_t msl_version = UINT32_MAX;
 	uint32_t vertex_input_mask = UINT32_MAX;
 	uint32_t fragment_output_mask = UINT32_MAX;
 	uint32_t spirv_specialization_constants_ids_mask = UINT32_MAX;
-	uint32_t is_compute = UINT32_MAX;
-	uint32_t needs_view_mask_buffer = UINT32_MAX;
```

Comment:
I dropped `is_compute` as we don't need to store it, and moved any boolean flags to a single `uint32_t`, so it is easier to add them in the future.